src/Microsoft.OData.Core/Json/ODataJsonWriterUtils.cs

```diff
@@ -356,5 +427,173 @@ private static void ODataResourceValueToString(StringBuilder sb, ODataResourceVa
 
             sb.Append("}");
         }
+
+        /// <summary>
+        /// Asynchronously writes an error message.
+        /// </summary>
+        /// <param name="jsonWriter">JSON writer.</param>
+        /// <param name="code">The code of the error.</param>
+        /// <param name="message">The message of the error.</param>
+        /// <param name="target">The target of the error.</param>
+        /// <param name="details">The details of the error.</param>
+        /// <param name="innerError">Inner error details that will be included in debug mode (if present).</param>
+        /// <param name="instanceAnnotations">Instance annotations for this error.</param>
+        /// <param name="maxInnerErrorDepth">The maximum number of nested inner errors to allow.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        private static async Task WriteErrorAsync(
+            IJsonWriterAsync jsonWriter,
+            string code,
+            string message,
+            string target,
+            IEnumerable<ODataErrorDetail> details,
+            ODataInnerError innerError,
+            IEnumerable<ODataInstanceAnnotation> instanceAnnotations,
+            int maxInnerErrorDepth)
+        {
+            Debug.Assert(jsonWriter != null, "jsonWriter != null");
+            Debug.Assert(code != null, "code != null");
+            Debug.Assert(message != null, "message != null");
+            Debug.Assert(instanceAnnotations != null, "instanceAnnotations != null");
+
+            // "error": {
+            await jsonWriter.StartObjectScopeAsync().ConfigureAwait(false);
+            await jsonWriter.WriteNameAsync(JsonConstants.ODataErrorName).ConfigureAwait(false);

```

Comment:
>ODataErrorName [](start = 58, length = 14)

This should write the jsonLight format, not the verbose format (i.e., it should use ODataErrorPropertyName).